corporate/views/upgrade.py

```diff
@@ -23,30 +20,32 @@
 )
 from corporate.models import CustomerPlan
 from zerver.decorator import require_organization_member, zulip_login_required
-from zerver.lib.request import REQ, has_request_variables
 from zerver.lib.response import json_success
 from zerver.lib.typed_endpoint import typed_endpoint
-from zerver.lib.validator import check_bool, check_int, check_string_in
 from zerver.models import UserProfile
 from zilencer.lib.remote_counts import MissingDataError
 
 billing_logger = logging.getLogger("corporate.stripe")
 
+# For Literal[] values see corporate/lib/stripe.py for reference.
+VALID_BILLING_MODALITY_VALUES = Literal["send_invoice", "charge_automatically"]
+VALID_BILLING_SCHEDULE_VALUES = Literal["annual", "monthly"]
+VALID_LICENSE_MANAGEMENT_VALUES = Literal["automatic", "manual"]
```

Comment:
Probably you should call these `VALID_BILLING_MODALITY_VALUES_TYPE` and still define them in `stripe.py`. Whatever we do, we definitely do not want to have duplicate definitions in different places; that's just asking for only one of them to get changed in the future.

Or maybe better would be to use `AfterValidator` like with existing settings.

Reply:
With this argument [this](https://github.com/zulip/zulip/blob/07b54e986bb849e6c776377e7d7bb448c1d63bac/corporate/views/support.py#L324) should be changed too, right ?

Reply:
Yes.